src/signTxn.c

```diff
@@ -402,11 +437,10 @@ void handleSignTxn(uint8_t p1, uint8_t p2, uint8_t *dataBuffer, uint16_t dataLen
 		FAIL("Cannot handle large data sent from host");
 	}
 
-	// Read the (partial) transaction
-	os_memmove(txndata, dataBuffer+dataOffset, txnLen);
+	// Read the (partial) transaction and
 	// Sign the txn and get message for confirmation display, all in ctx.
 	// Signature will not go back to host until message display + approval.
-	sign_deserialize_stream(txndata, txnLen, hostBytesLeft);
+	sign_deserialize_stream(dataBuffer + dataOffset, txnLen, hostBytesLeft);
```

Comment:
Return value should be checked here, to avoid calling UX_DISPLAY and display uninitialized data